### PR TITLE
Add the option to run the tests using the bytecode interpreter 

### PR DESCRIPTION
**Pull request by alimpfard**
Also a very random drive-by fix lol

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -421,6 +440,12 @@ def main() -> None:
         metavar="PATH",
         help="path to the 'libjs-test262-runner' binary",
     )
+    parser.add_argument(
+        "-b",
+        "--use-bytecode",
+        action="store_true",
+        help="Use the bytecode interpreter to run the tests",
+    )
```
Comment (linusg) [inline main.py:448]: needs an update of the `--help` dump in the readme please :)

File: main.py [hunk 2]
```diff
@@ -437,7 +462,7 @@ def main() -> None:
     parser.add_argument(
         "-c",
         "--concurrency",
-        default=CPU_COUNT,
+        default=CPU_COUNT * 5,
         type=int,
         help="number of concurrent workers (defaults to number of CPU cores)",
```
Comment (linusg) [inline main.py:467]: maybe...

```suggestion
        help="number of concurrent workers (defaults to number of CPU cores * 5)",
```

**Review by linusg — COMMENTED** [review]
noice!

File: src/main.cpp [hunk 3]
```diff
@@ -20,6 +24,8 @@
 #include <stdio.h>
 #include <unistd.h>
 
+static bool s_use_bytecode = false;
```
[the remark this answers is not in the document]
Comment (alimpfard) [reply src/main.cpp:27]: I was just expecting to use it in `run_program()`, but I ended up using `IsSame<...>` instead.
I'll move it into `main`.

File: main.py (hunk 1 above)
Comment (alimpfard) [reply main.py:448]: Sure~

File: run_all_and_update_results.py [hunk 4]
```diff
@@ -130,6 +130,18 @@ def main() -> None:
     )
     libjs_test262_results = libjs_test262_output["results"]["test"]["results"]
 
+    print("Running test262 with the bytecode interpreter...")
+    libjs_test262_bytecode_output = json.loads(
+        # This is not the way either, but I can't be bothered to fix the one above and _then_ copy it. :^)
```
[the remark this answers is not in the document]
Comment (alimpfard) [reply run_all_and_update_results.py:135]: tech debt, where I loan from your loan :P

File: run_all_and_update_results.py [hunk 5]
```diff
@@ -154,6 +166,19 @@ def main() -> None:
                     "runner_exception": libjs_test262_results["RUNNER_EXCEPTION"],
                 },
             },
+            "test262-bytecode": {
+                "results": {
```
Comment (alimpfard) [inline run_all_and_update_results.py:170]: /me forgets "duration" like an idiot
:(

**Pull request merged**